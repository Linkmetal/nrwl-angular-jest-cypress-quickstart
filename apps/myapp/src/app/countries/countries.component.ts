import { Component, OnInit } from '@angular/core';
import { CountriesService } from './countries.service';
import { Observable } from 'rxjs';
import { Country } from './country';

@Component({
  selector: 'myapp-countries',
  templateUrl: './countries.component.html',
  styleUrls: ['./countries.component.css']
})
export class CountriesComponent implements OnInit {

  countries$: Observable<Country[]>;

  constructor(private countriesService: CountriesService) { }

  ngOnInit() {
    this.countries$ = this.countriesService.getCountries();
  }

}
