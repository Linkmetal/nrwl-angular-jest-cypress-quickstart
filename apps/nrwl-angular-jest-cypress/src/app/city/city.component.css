.city {
    font: 200 20px/1.5 Helvetica, Verdana, sans-serif;
    position: fixed; 
    top: 20px; 



    right: 0;
    width: 300px;
    min-height: 100px;
    background: hsla(80, 90%, 40%, 0.7);
    color: white;
    font-weight: bold;
    font-size: large;
    text-align: left;
    border: solid hsla(80, 90%, 40%, 0.5);
    border-right: none;
    padding: 0.5em 0.5em 0.5em 2.5em;
    box-shadow: 0 1px 3px black;
    border-radius: 3em 0.5em 0.5em 3em;
}