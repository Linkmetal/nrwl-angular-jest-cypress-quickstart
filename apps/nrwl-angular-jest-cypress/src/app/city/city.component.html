<div style="position: fixed; top: 0; right: 20px;">
  <h1 *ngIf="cityService.getCity() | async as city">
    {{ city }}
  </h1>
</div>

