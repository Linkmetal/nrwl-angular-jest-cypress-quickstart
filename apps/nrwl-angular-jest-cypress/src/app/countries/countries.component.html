<div>
  <ul>
    <li *ngFor="let country of countries$ | async" (click)="selectCountry(country.city)">{{ country.country }}</li>
  </ul>
</div>

