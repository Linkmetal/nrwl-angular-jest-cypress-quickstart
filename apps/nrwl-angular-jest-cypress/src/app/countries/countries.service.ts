import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class CountriesService {

  private readonly url = 'http://localhost:3000/countries';

  constructor(private httpClient: HttpClient) { }

  public getCountries() {
    return this.httpClient.get(this.url);
  }
}
