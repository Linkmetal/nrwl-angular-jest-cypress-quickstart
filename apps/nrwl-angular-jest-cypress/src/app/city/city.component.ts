import { Component, OnInit } from '@angular/core';
import { CityService } from '../city.service';

@Component({
  selector: 'nrwl-angular-jest-cypress-city',
  templateUrl: './city.component.html',
  styleUrls: ['./city.component.css']
})
export class CityComponent {
  constructor(public cityService: CityService) { }
}
