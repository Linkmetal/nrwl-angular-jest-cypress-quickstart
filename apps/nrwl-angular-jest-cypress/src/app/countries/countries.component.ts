import { Component, OnInit } from '@angular/core';
import { CountriesService } from './countries.service';
import { CityService } from '../city.service';

@Component({
  selector: 'nrwl-angular-jest-cypress-countries',
  templateUrl: './countries.component.html',
  styleUrls: ['./countries.component.css']
})
export class CountriesComponent implements OnInit {

  countries$: any;

  constructor(private countriesService: CountriesService, private cityService: CityService) { }

  ngOnInit() {
    this.countries$ = this.countriesService.getCountries();
  }

  selectCountry(city: string) {
    this.cityService.city.emit(city);
  }
}
