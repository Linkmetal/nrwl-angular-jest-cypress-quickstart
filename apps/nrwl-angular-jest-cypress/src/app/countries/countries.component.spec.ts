import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { CountriesComponent } from './countries.component';
import { CountriesService } from './countries.service';
import { CityService } from '../city.service';
import { of } from 'rxjs';
import { By } from '@angular/platform-browser';

jest.mock('./countries.service');

describe('CountriesComponent', () => {
  let component: CountriesComponent;
  let fixture: ComponentFixture<CountriesComponent>;

  let countriesService: CountriesService;
  let cityService: CityService;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ CountriesComponent ],
      providers: [CountriesService, CityService]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    countriesService = TestBed.get(CountriesService);
    cityService = TestBed.get(CityService);

    fixture = TestBed.createComponent(CountriesComponent);
    component = fixture.componentInstance;

    spyOn(countriesService, 'getCountries').and.returnValue(of([
      { country: 'Spain', city: 'Madrid' },
      { country: 'France', city: 'Paris'}
    ]));

    spyOn(cityService, 'emitCity');

    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
  
  it('should render component', () => {
    expect(fixture).toMatchSnapshot();
  });
  
  it('should show a list of elements', () => {
    const firstElement = fixture.debugElement.query(By.css('li:first-child'));
    firstElement.nativeElement.click();
    expect(cityService.emitCity).toHaveBeenCalledWith('Madrid');
  });
});
