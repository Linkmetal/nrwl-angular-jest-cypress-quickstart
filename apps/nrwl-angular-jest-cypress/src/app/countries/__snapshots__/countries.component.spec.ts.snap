// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountriesComponent should render component 1`] = `
<nrwl-angular-jest-cypress-countries
  cityService={[Function CityService]}
  countries$={[Function Observable]}
  countriesService={[Function CountriesService]}
>
  <div>
    <ul>
      
      <li>
        Spain
      </li>
      <li>
        France
      </li>
    </ul>
  </div>
</nrwl-angular-jest-cypress-countries>
`;
